src/config/mod.rs

```diff
@@ -213,15 +213,44 @@ impl PartialConfig {
     }
 }
 
+fn check_semver_version(required: &str, actual: &str) -> bool {
+    let mut required_version = match semver::VersionReq::parse(required) {
+        Ok(r) => r,
+        Err(e) => {
+            eprintln!("Error: failed to parse required version: {}", e);
+            return false;
+        }
+    };
+    let actual_version = match semver::Version::parse(actual) {
+        Ok(v) => v,
+        Err(e) => {
+            eprintln!("Error: failed to parse current version: {}", e);
+            return false;
+        }
+    };
+
+    required.split(',').enumerate().for_each(|(i, required)| {
+        let Some(comparator) = required_version.comparators.get_mut(i) else {
+            return;
+        };
+
+        if !required.contains('^') && comparator.op == semver::Op::Caret {
+            comparator.op = semver::Op::Exact;
+        }
+    });
```

Comment:
Could you add a comment somewhere to this section to explain that this is used to workaround the default behavior of the `semver` crate. I think having that context captured in a comment for future readers of the code would be very useful.

Reply:
Should the check change to this?

```diff

-        if !required.contains('^') && comparator.op == semver::Op::Caret {
+        if !required.starts_with'^') && comparator.op == semver::Op::Caret {
            comparator.op = semver::Op::Exact;
        }
```

also, to avoid any confusion between the `required` passed to the `check_semver_version` and the `required` in the body of the closure maybe one of those variables should be renamed?

Reply:
done @ytmimi. sorry for the delay 😅

let me know if you have any additional feedback on this